daemon/common_freebsd.go

```diff
@@ -90,36 +92,53 @@ run_rc_command "$1"
 	rcConfig := `netclient="YES"
 netclient_args="daemon"`
 
+	slog.Info("Installing netclient service files")
 	rcbytes := []byte(rcFile)
-	if !ncutils.FileExists("/etc/rc.d/netclient") {
-		err := os.WriteFile("/etc/rc.d/netclient", rcbytes, 0744)
-		if err != nil {
-			return err
-		}
-		rcConfigbytes := []byte(rcConfig)
-		if !ncutils.FileExists("/etc/rc.conf.d/netclient") {
-			err := os.WriteFile("/etc/rc.conf.d/netclient", rcConfigbytes, 0644)
-			if err != nil {
-				return err
-			}
-			start()
-			return nil
-		}
+	if err := os.WriteFile("/etc/rc.d/netclient", rcbytes, 0755); err != nil {
+		return err
+	}
+	rcConfigbytes := []byte(rcConfig)
+	if err := os.WriteFile("/etc/rc.conf.d/netclient", rcConfigbytes, 0644); err != nil {
+		return err
 	}
+	slog.Info("starting deamon")
+	start()
 	return nil
 }
 
 func start() error {
-	return service("start")
+	if status() {
+		return errors.New("service already running")
+	}
+	service("start")
+	if status() {
+		return nil
+	}
+	return errors.New("failed to start service")
 }
 
 func stop() error {
-	return service("stop")
+	if !status() {
+		return errors.New("service not running")
+	}
+	service("stop")
+	if status() {
```

Comment:
is it a good idea to immediately check for status here after a stop signal? 
I believe the `service netclient stop` command sends a kill signal to the daemon and exits, won't the daemon take a second or two to halt? then this status check will always return an error

Reply:
the freebsd service xxxx stop command runs in a loop.  It does not exit the loop until the pid of the process being killed is no longer present.